$OpenBSD: patch-lib_dictBuilder_cover_c,v 1.1 2018/04/04 16:10:02 bket Exp $

zstd doesn't play nice with libc's qsort(). Reason is that qsort() is not
guaranteed to be stable. Proposed fix is to use libc's mergesort() instead.

https://github.com/facebook/zstd/issues/1088

Index: lib/dictBuilder/cover.c
--- lib/dictBuilder/cover.c.orig
+++ lib/dictBuilder/cover.c
@@ -583,7 +583,7 @@ static int COVER_ctx_init(COVER_ctx_t *ctx, const void
     }
     /* qsort doesn't take an opaque pointer, so pass as a global */
     g_ctx = ctx;
-    qsort(ctx->suffix, ctx->suffixSize, sizeof(U32),
+    mergesort(ctx->suffix, ctx->suffixSize, sizeof(U32),
           (ctx->d <= 8 ? &COVER_strict_cmp8 : &COVER_strict_cmp));
   }
   DISPLAYLEVEL(2, "Computing frequencies\n");
